ecco_v4_py/resample_to_latlon.py

```diff
@@ -82,14 +82,43 @@ def resample_to_latlon(orig_lons, orig_lats, orig_field,
         raise TypeError('orig_lons and orig_lats variable either a DataArray or numpy.ndarray. \n'
                 'Found type(orig_lons) = %s and type(orig_lats) = %s' %
                 (type(orig_lons), type(orig_lats)))
-
+    
     if type(orig_field) == xr.core.dataarray.DataArray:
         orig_field = orig_field.values
     elif type(orig_field) != np.ndarray and \
          type(orig_field) != np.ma.core.MaskedArray :
         raise TypeError('orig_field must be a type of DataArray, ndarray, or MaskedArray. \n'
                 'Found type(orig_field) = %s' % type(orig_field))
 
+    ## Modifications to allow time and depth dimensions (DS, 2023-04-20)    
+    # Collapse any non-horizontal dimensions into a single, final dimension:
+
+    # Get shape of orig_lats, then difference with orig_field
+    n_horiz_dims=len(orig_lats.shape)
+    n_total_dims=len(orig_field.shape)
+    n_extra_dims=n_total_dims-n_horiz_dims
+    horiz_dim_shape=orig_lats.shape # e.g. [13,90,90]    
+    if ( (n_extra_dims>0) & (np.prod(orig_field.shape) > np.prod(horiz_dim_shape) ) ):
```

Comment:
Just for my own understanding, shouldn't `n_extra_dims>0` be enough? Is there a reason for the second condition?

Reply:
Originally I had it with just `n_extra_dims>0` but things fail if there is a trivial extra dimension of length 1. The second condition is to check the extra dimension actually contains data/is "meaningful" in some way.